src/test/unit_tests/test_nc_iam_basic_integration.js

```diff
@@ -0,0 +1,234 @@
+/* Copyright (C) 2024 NooBaa */
+/* eslint-disable max-statements */
+'use strict';
+
+const path = require('path');
+const _ = require('lodash');
+const mocha = require('mocha');
+const assert = require('assert');
+const fs_utils = require('../../util/fs_utils');
+const { TMP_PATH, generate_nsfs_account, get_new_buckets_path_by_test_env, generate_iam_client,
+         get_coretest_path } = require('../system_tests/test_utils');
+const { ListUsersCommand, CreateUserCommand, GetUserCommand, UpdateUserCommand, DeleteUserCommand,
+        ListAccessKeysCommand, CreateAccessKeyCommand, GetAccessKeyLastUsedCommand,
+        UpdateAccessKeyCommand, DeleteAccessKeyCommand } = require('@aws-sdk/client-iam');
+const { ACCESS_KEY_STATUS_ENUM } = require('../../endpoint/iam/iam_constants');
+
+const coretest_path = get_coretest_path();
+const coretest = require(coretest_path);
+const setup_options = { should_run_iam: true, https_port_iam: 7005, debug: 5 };
+coretest.setup(setup_options);
+const { rpc_client, EMAIL, get_current_setup_options, stop_nsfs_process, start_nsfs_process } = coretest;
+
+const CORETEST_ENDPOINT_IAM = coretest.get_iam_https_address();
+
+const config_root = path.join(TMP_PATH, 'test_nc_iam');
+// on NC - new_buckets_path is full absolute path
+// on Containerized - new_buckets_path is the directory
+const new_bucket_path_param = get_new_buckets_path_by_test_env(config_root, '/');
+
+let iam_account;
+
+mocha.describe('IAM basic integration tests - happy path', async function() {
```

Comment:
do we need also integration tests between both IAM and S3 services?
for example, I would expect a created user can do I/O etc

Reply:
okay, so please add a sub-task